### ISSUE TITLE
New viewer: playing and browsing give a different result

### ISSUE THREAD

**Issue by shlublu** [mapillary/mapillary_issues#1326]
Playing sequences and browsing them give a different result:

Let's take this sequence as an example: http://www.mapillary.com/map/im/rfOq5vO_VMt1gun30qdzOg/photo

1- If I click on 'play', I'll obtain a kind of panoramic view

2- If I click on 'next', I'll have a 'forward' button available, but no 'right' and 'left' buttons. And this will lead me straight to another sequence. So I won't have a chance to browse the panoramic view

3- Let's say I did as described in 2- until I reach the stairway roughly 100m away. Then I click on 'backward', expecting to follow the same path reverse, but I'll stay on the sequence I was routed to. I'll never reach the initial picture (linked at the top of this post) again.

This gives a strange impression. As a user I feel lost, somehow. It's also annoying during blurring sessions as it makes difficult to determine whether a whole sequence was properly blurred, without missing anything.

**Comment by oscarlorentzon** [comment]
You are right @shlublu , playing (and dragging) is done within a sequence whereas moving around by clicking the navigation arrows can take you to an image in another sequence. We determine the best match for movement in each direction which in this case makes it hard to get back to the sequence you started from once you are on the second sequence. 

There is one image in the second sequence from which you can get back to the first (click the right arrow): http://www.mapillary.com/map/im/kkl_YPIYkIomEgDl9JeQGw/photo

Currently, the best way to navigate to a specific image in a sequence is the following (we aim to simplify this).
1. Browse to another image in the sequence.
2. Click and hold the left mouse button somewhere on the image to start a dragging session. Drag slowly up or down on the image to move to the previous or next image in the sequence.
3. Release the mouse button when you are at the sequence image you intended to edit.

Thanks for reporting this! Examples of places where it is hard to navigate will help us improve the behaviour.

**Comment by shlublu** [comment]
Thanks for the explanation and the tips'n'tricks @oscarlorentzon. I'll do that way when blurring or just browsing a sequence now.

**Comment by oscarlorentzon** [comment]
@shlublu There is now an option in the Edit accordion (visible when logged in) to lock movement to the sequence of the currently viewed photo. When locked, the only navigation alternatives are then the previous and the next photo in the sequence (use navigation arrows in UI or keyboard up and down buttons). 

This should simplify the editing process when going through a specific sequence.

**Comment by shlublu** [comment]
Good idea! It's funny though because I cannot reproduce the scenario above even if my sequence is unlocked. As if the connection between the two sequences was lost.

**Comment by oscarlorentzon** [comment]
I still have the same behavior as before. From http://www.mapillary.com/map/im/rlS18yPhF10-dv6Xg9K-gA/photo in the first sequence I navigate to the second sequence by going forward. From http://www.mapillary.com/map/im/kkl_YPIYkIomEgDl9JeQGw/photo in the second sequence I navigate to the first sequence by going right.

The problem with the first sequence is that there is not much (or any) overlap between the images since they are taken in very different angles. Therefore it is not possble to reconstruct the images in the first sequence to get a good relative rotation and position with respect to the other images.

**Comment by shlublu** [comment]
Sorry it's fine. The problem was between the chair and keyboard...

**Comment by oscarlorentzon** [comment]
In `mapillary-js` we have separated the spatial navigation and the sequence navigation completely so that it should be easier to understand how to navigate either within a sequence or randomly through space. 

Still, being able to move back to the same image you came from is important.

**Comment by ybkuang** [comment]
mapillary_issues is no longer an active repository.
Let’s continue discussing features in our forum! We will import this issue as topic to the Mapillary Forum in the coming weeks. We’ll notify here once it’s been done. See you there!
